### ISSUE TITLE
v1.0.0-rc.6 DB Auth getToken unsetting cookies

### ISSUE THREAD

**Issue by pi0neerpat** [redwoodjs/redwood#4710]
When using the new the new cookie settings in dbAuth, `getToken` now returns response cookies, where it was not doing so before. Since the getToken response cookie can be null, such as during logging in, it can unset the cookie, thus logging the user out.

```js
  cookie: {
      HttpOnly: true,
      Path: '/',
      SameSite: 'Strict',
      Secure: true,
      // Domain: 'example.com',
    },
```

**Comment by thedavidprice** [comment]
Thanks @pi0neerpat! Looping in @cannikin

**Comment by cannikin** [comment]
Hmm...using this with oauth isn't really one of the use cases dbAuth is meant to cover, it's really just meant for your old fashioned email/password login box right on the page.

In your steps above, step 2 actually completes *after* step 3, correct? So if `completeOauthLogin` is setting the cookie *after* `getToken` has set it to expire it, why does not it not stick after that?

Is there anything weird with moving between different domains? Although if it was working before I'm guessing not? The reason we added this config to the function is because people with web and api sides on different domains weren't able to use dbAuth—CORS and `SameSite` config wouldn't allow the cookie to be sent up to the API side, so even though the login function says "you're logged in, here's the cookie" the browser would refuse to save the cookie and/or send the cookie back up to `getToken`.  We added a whole CORS doc for the config changes that are required: https://redwoodjs.com/docs/cors

**Comment by cannikin** [comment]
There is one behavior that's different with the new cookie config: the `Secure` attribute on the cookie is set to whatever value you have in your config, whereas before it looked at `NODE_ENV` and only set it to `true` if it was *not* `"development"`. But, it appears that browsers have an exception where you *can* set a `Secure` cookie over `http` as long as the domain is `localhost`. So we just defaulted it to `true` in the config template.

**Comment by pi0neerpat** [comment]
It is indeed a problem with `Secure: true` over http. I wasn't showing warning level logs, which is why I didnt catch this earlier.

![image](https://user-images.githubusercontent.com/35622595/157784753-cab3e4d4-fe83-4144-9631-413e8e1559de.png)

**Comment by cannikin** [comment]
Sweet! So you can put that same check for NODE_ENV in the config and you should be good! Or use localhost.

**Comment by pi0neerpat** [comment]
Oh I just re-read your comment. I see now its b/c I'm using `0.0.0.0` instead of localhost, not related to the browser. I wonder if its worth adding a comment in the docs to help other devs https://redwoodjs.com/docs/authentication#cookie-config

**Comment by rkmitra1** [comment]
In QA testing the tutorial, I may have run into this problem.

Problem: dbAuth: self hosted works on Chrome but does not work correctly on Safari.

Notes:
in api/src/functions/auth.js

DOES NOT WORK
`  cookie: {
      HttpOnly: true,
      Path: '/',
      SameSite: 'Strict',
      Secure: true,`

WORKS
`  cookie: {
      HttpOnly: true,
      Path: '/',
      SameSite: 'Strict',
      Secure: false,`


My testing environment:
node v16.11.1
os: Mac 12.2
tutorial section: https://learn.redwoodjs.com/docs/tutorial/authentication

Basically using the tutorial blog app.
Works on chrome: login takes me back to home page, logged in. I can access admin sections
Does not work on safari: login displays welcome message, but stays on login page. I am not logged in

**Comment by cannikin** [comment]
Are you accessing your page via `localhost` or through an IP address? Chrome allows an exception that you *can* set a cookie as `Secure` when accessing via `localhost`...

**Comment by cannikin** [comment]
If that doesn't help, can you try changing you cookie config to:

```javascript
cookie: {
  HttpOnly: true,
  Path: '/',
  SameSite: 'Strict',
  Secure: process.env.NODE_ENV !== 'development' ? true : false,
},
```

**Comment by rkmitra1** [comment]
I am accessing using localhost.

`  Secure: process.env.NODE_ENV !== 'development' ? true : false,
`
Your change fixes the problem.

**Comment by cannikin** [comment]
Thanks for the update. I'm going to make that the default in the dbAuth function template going forward. PR opened!
